src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -279,27 +279,36 @@ public bool HasNoResults
         }
 
         /// <summary>
-        /// Checks if the user already has a package by the given name.
+        /// Checks if the package can be installed.
         /// </summary>
         /// <param name="packageName"></param>
         /// <returns></returns>
-        internal bool UserAlreadyHasPackageInstalled(string packageName)
+        internal bool CanInstallPackage(string name)
         {
-            PackageManagerExtension packageManagerExtension = PackageManagerClientViewModel.PackageManagerExtension;
-            
-            // This check fires before the package is added to PackageLoader.LocalPackages collection.
-            // Therefore, we need to check both the names of downloaded packages and recent downloads.
-
-            List<string> loadedPackageNames = packageManagerExtension.PackageLoader.LocalPackages
-                .Where(x => x.LoadState.State == PackageLoadState.StateTypes.Loaded)
-                .Select(x => x.Name)
-                .ToList();
-
-            List<string> recentlyDownloadedPackageNames = Downloads
-                .Select(x => x.Name)
-                .ToList();
+            // Return true if there are no matching non built-in packages
+            return false == PackageManagerClientViewModel.PackageManagerExtension.PackageLoader.LocalPackages
```

Comment:
how about `return !PackageManagerClientViewModel.PackageManagerExtension.PackageLoader.LocalPackages...`

Reply:
why not...I wanted it to be very explicit just because it can be a bit confusing...